### DISCUSSION TITLE
Could not find the Ghostty resources directory after installing to $HOME/.local

### DISCUSSION THREAD

**Discussion by melhindi** [ghostty-org/ghostty#3500] [Feature Requests, Ideas] [answered]
Thanks for this great project! :)
I had a small issue during installation on my linux box:
```
lsb_release -a
LSB Version:    n/a
Distributor ID: EndeavourOS
Description:    EndeavourOS Linux
Release:        rolling
Codename:       rolling
```

```
uname -a
Linux thinkpad-x1 6.12.6-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 19 Dec 2024 21:29:01 +0000 x86_64 GNU/Linu
```

I have installed ghostty with the following command:
` zig build -p $HOME/.local -Doptimize=ReleaseFast`

Following XDG_ variables are set:
```
XDG_RUNTIME_DIR    XDG_SEAT           XDG_SESSION_CLASS  XDG_SESSION_ID     XDG_SESSION_TYPE   XDG_VTNR
```

## Observation
Executing ` $HOME/.local/bin/ghostty +list-themes` results in the following:
```
Could not find the Ghostty resources directory. Please ensure that Ghostty is installed correctly.
No themes found, check to make sure that the themes were installed correctly.%
```

Similarly:
```
$HOME/.local/bin/ghostty --theme=xcodelight
info: ghostty version=1.0.0-main+4b4d4062
info: ghostty build optimize=ReleaseFast
info: runtime=apprt.Runtime.gtk
info: font_backend=font.main.Backend.fontconfig_freetype
info: dependency harfbuzz=8.4.0
info: dependency fontconfig=21402
info: renderer=renderer.OpenGL
info: libxev backend=main.Backend.io_uring
info(os): setlocale from env result=LC_CTYPE=en_US.UTF-8;LC_NUMERIC=de_DE.UTF-8;LC_TIME=de_DE.UTF-8;LC_COLLATE=en_US.UTF-8;LC_MONETARY=de_DE.UTF-8;LC_MESSAGES=en_US.UTF-8;LC_PAPER=de_DE.UTF-8;LC_NAME=de_DE.UTF-8;LC_ADDRESS=de_DE.UTF-8;LC_TELEPHONE=de_DE.UTF-8;LC_MEASUREMENT=de_DE.UTF-8;LC_IDENTIFICATION=de_DE.UTF-8
info(gtk): GTK version build=4.16.7 runtime=4.16.7
info: reading configuration file path=/home/$USER/.config/ghostty/config
info(config): default shell source=env value=/usr/bin/zsh
warning(gtk): configuration error: theme "xcodelight" not found, tried path "/home/$USER/.config/ghostty/themes/xcodelight"
info(sentry): sentry envelope does not contain crash, discarding
```

## Expected behavior
Ghostty uses `$HOME/.local/share/ghostty` (install prefix dir) instead of ` $HOME/.config/ghostty` (config folder) to search for resources

## Workarounds
- Installing to a different prefix, e.g., `zig build -p $HOME/mydir -Doptimize=ReleaseFast` works as expected
- setting `GHOSTTY_RESOURCES_DIR` manually works as expected, e.g. ` GHOSTTY_RESOURCES_DIR=$HOME/.local/share/ghostty/ $HOME/.local/bin/ghostty +list-themes`

Let me know if there are any other infos/details that I should provide

**Comment by mitchellh** [comment]
Convert to discussion for issue triage.

**Comment by asilvam133** [comment]
Moving/copying `~/.local/share/ghostty/themes` to `$HOME/.config/ghostty/themes` works for the moment.

**Comment by cbusoi** [comment]
I have a similar issue under Fedora 41. 
When starting ghostty from another terminal (ptyxis) everything worked as expected. 

After starting ghossty over the file explorer I receive the following message:

`theme "Dracula+" not found, tried path "/home/username/.config/ghostty/themes/Dracula+"`

Looks like when not started over the cli ghostty expects the themes at another location...

**Reply by mBuergi86** [reply]
Not working!

**Comment by melhindi** [comment]
quick update:
I added a simple print statement in `src/os/resourcedir.zig` to understand which paths are being checked.
I think the problem is that my `$HOME/.local/bin` folder (as determined by `std.fs.selfExePath()`) is a symbolic link:
```
 check dir: /home/$USER/git/stowfiles/bin/.local/bin
 check dir: /home/$USER/git/stowfiles/bin/.local
 check dir: /home/$USER/git/stowfiles/bin
 check dir: /home/$USER/git/stowfiles
 check dir: /home/$USER/git
 check dir: /home/$USER
 check dir: /home
 check dir: /
```

Obviously, there is no `share/ghostty` folder in `git/stowfiles`. During installation, the `resource_dir` containing the `themes` folder has been installed to `/home/$USER/.local/share` (`/home/$USER/.local` is not sym-linked)

Not sure what would be the correct "fix" in this case. At the moment, I can imagine 3 options:
1) Advise the user not to install to symlinked folders (i.e., `/$HOME/.local/bin` should not be sym-linked)
2) During installation, make sure to follow sym-links and install the resources_dir accordingly
3) Change the code to not follow sym-links when searching for resources_dir

**Comment by melhindi** [comment]
I just checked the docu for `std.fs.selfExePath`: https://ziglang.org/documentation/0.13.0/std/#std.fs.selfExePath
It says that it **Follows symlinks.**.
Since the installation does not follow symlinks, personally, I think we should not be following symlinks in `resourcesDir()`, too.
However, with only a quick check, I couldn't find an alternative function that does not follow symlinks.

**Comment by mBuergi86** [comment]
Me too not work!

**Reply by mBuergi86** [reply]
Got it! Let’s tackle the **"Could not find the Ghostty resources directory"** issue systematically with the cleanest solution. Here's a step-by-step guide:

---

## Correct and Complete Solution for Ghostty

### 1. **Clone the Repository**
   First, clone the repository to your local machine:
   ```bash
   git clone https://github.com/mbadolato/iTerm2-Color-Schemes.git
   ```

### 2. **Navigate to the Ghostty Directory**
   Go into the Ghostty-specific folder:
   ```bash
   cd iTerm2-Color-Schemes/ghostty
   ```

### 3. **Copy Themes to the Config Directory**
   Copy the themes into Ghostty’s configuration directory:
   ```bash
   mkdir -p ~/.config/ghostty/themes
   cp * ~/.config/ghostty/themes/
   ```

### 4. **Remove the Cloned Repository**
   Clean up by deleting the cloned repository:
   ```bash
   cd ~
   rm -rf iTerm2-Color-Schemes
   ```

### 5. **List Available Themes**
   Test if Ghostty can locate the themes:
   ```bash
   ghostty +list-themes
   ```

---

### If the Error Persists: Manual Path Configuration

If the error **"Could not find the Ghostty resources directory"** still occurs, Ghostty may not automatically recognize the default path. Manually set the resources directory:

1. **Export the Resource Path**
   Temporarily set the path:
   ```bash
   export GHOSTTY_RESOURCES=~/.config/ghostty/themes
   ```

2. **Make the Export Persistent**
   Add the export command to your shell configuration file:
   - For **zsh**:
     ```bash
     echo 'export GHOSTTY_RESOURCES=~/.config/ghostty/themes' >> ~/.zshrc
     source ~/.zshrc
     ```

3. **Re-run the Theme Listing**
   Try again to list the available themes:
   ```bash
   ghostty +list-themes
   ```
But I’m still getting the following error:

"Could not find the Ghostty resources directory. Please ensure that Ghostty is installed correctly."

**Reply by unnask** [reply]
This fixed it. Mine stopped giving me a theme after an update. Thanks!

**Reply by Mr-Sunglasses** [reply]
This works, Thanks for the solution.

**Comment by nduartech** [comment]
So I'm specifically on EndeavourOS and am using chezmoi to manage my config. I faced this issue when using ghostty-git, but it was resolved when I fell back to the ghostty package. Not sure if any of that helps

**Reply by 00-kat** [reply]
This discussion is about installing to `~/.local`, while `ghostty-git` installs to `/usr`.  And the issue was likely because the AUR package was split a little while ago, and both yay and paru didn't install the two new packages, `ghostty-terminfo-git` and `ghostty-shell-integration-git`.

**Comment by pete-willard** [comment]
In my case themes broke after an update too without me doing any tinkering. I am 100% certain listing themes worked at one point. I don't even have a `~/.local/share/ghostty` directory atm. Neither a `~/.config/ghostty/themes` one. I'm on Manjaro btw

**Reply by BlueManCZ** [reply]
This looks like a regression in 1.1.0. I workarounded it by creating a symling to `/usr/share/ghostty/themes`.
```
ln -s /usr/share/ghostty/themes ~/.config/ghostty
```

**Comment by pktpls** [comment]
I'm on Fedora 41, Ghostty installed to `$HOME/.local`, and it has been finding the themes fine. Except today it suddenly decided that it couldn't anymore. This happened once about 2 months ago, back then I rebuilt from a current git commit, and it worked again.

I don't have Zig 0.14 yet so I can't rebuild at the moment, but I tried the symlink to `$HOME/.config/themes` and that works.

**Comment by JVPH** [comment]
Im on fedora 42, the themes show in the +list-themes selector, but won't work if set in the config file.

I have tried:
- symlink
- copying the contents from /usr/share to .config/ghostty/themes
- exposing the resource path.

None of it has worked.

**Reply by Mr-Sunglasses** [reply]
> Got it! Let’s tackle the **"Could not find the Ghostty resources directory"** issue systematically with the cleanest solution. Here's a step-by-step guide:
> 
> ## Correct and Complete Solution for Ghostty
> ### 1. **Clone the Repository**
> First, clone the repository to your local machine:
> 
> ```shell
> git clone https://github.com/mbadolato/iTerm2-Color-Schemes.git
> ```
> 
> ### 2. **Navigate to the Ghostty Directory**
> Go into the Ghostty-specific folder:
> 
> ```shell
> cd iTerm2-Color-Schemes/ghostty
> ```
> 
> ### 3. **Copy Themes to the Config Directory**
> Copy the themes into Ghostty’s configuration directory:
> 
> ```shell
> mkdir -p ~/.config/ghostty/themes
> cp * ~/.config/ghostty/themes/
> ```
> 
> ### 4. **Remove the Cloned Repository**
> Clean up by deleting the cloned repository:
> 
> ```shell
> cd ~
> rm -rf iTerm2-Color-Schemes
> ```
> 
> ### 5. **List Available Themes**
> Test if Ghostty can locate the themes:
> 
> ```shell
> ghostty +list-themes
> ```
> 
> ### If the Error Persists: Manual Path Configuration
> If the error **"Could not find the Ghostty resources directory"** still occurs, Ghostty may not automatically recognize the default path. Manually set the resources directory:
> 
> 1. **Export the Resource Path**
>    Temporarily set the path:
>    ```shell
>    export GHOSTTY_RESOURCES=~/.config/ghostty/themes
>    ```
> 2. **Make the Export Persistent**
>    Add the export command to your shell configuration file:
>    
>    * For **zsh**:
>      ```shell
>      echo 'export GHOSTTY_RESOURCES=~/.config/ghostty/themes' >> ~/.zshrc
>      source ~/.zshrc
>      ```
> 3. **Re-run the Theme Listing**
>    Try again to list the available themes:
>    ```shell
>    ghostty +list-themes
>    ```
> 
> But I’m still getting the following error:
> 
> "Could not find the Ghostty resources directory. Please ensure that Ghostty is installed correctly."

Hi @JVPH I've tested the above method in fedora 42 and it is working for me. Try it and let me know if it works for you

**Reply by JVPH** [reply]
Yes, that's the one I just tried. Doesn't work for some reason :thinking:

**Reply by Mr-Sunglasses** [reply]
> Yes, that's the one I just tried. Doesn't work for some reason 🤔

Oh, did you check if  `GHOSTTY_RESOURCES` is set ?

Also please make sure you extract the themes in the right path.

**Comment by mitchellh** [comment, marked as answer]
I'm going to close this discussion. I think its far too stale at this point and we need an up to date direct reproduction of what we're trying to change. This is devolving into a general purpose "resources dir can't be found" forum.

**Comment by crivotz** [comment]
I had similar problem with snap version on 1.2.0 version and I do 
`ln -s /snap/ghostty/current/share/ghostty/themes .config/ghostty`

**Comment by binary-signal** [comment]
For macos this worked, when installed via brew

export GHOSTTY_RESOURCES=/Applications/Ghostty.app/Contents/Resources/ghostty/themes
